lsp-mode.el

```diff
@@ -5646,29 +5659,32 @@ perform the request synchronously."
            (lsp-request "workspace/symbol" `(:query ,pattern))))
 
 (defun lsp--get-symbol-to-rename ()
-  "Get symbol to rename and placeholder at point."
-  (if (let ((rename-provider (or (lsp--capability :renameProvider)
-                                 (-some-> (lsp--registered-capability "textDocument/rename")
-                                   (lsp--registered-capability-options)))))
-        (lsp:rename-options-prepare-provider? rename-provider))
-      (-when-let (response (lsp-request "textDocument/prepareRename"
-                                        (lsp--text-document-position-params)))
+  "Get symbol to rename and placeholder at point.
+Returns a cons (SYMBOL-TO-RENAME . PLACEHOLDER)."
+  (unless (lsp-feature? "textDocument/rename")
+    (lsp--fatal-error "The connected server(s) doesn't support renaming"))
+  ;; Do we have prepareRename? If so, we can make use of it as much as possible.
+  (if (lsp-feature? "textDocument/prepareRename")
+      (when-let ((prepare
+                  (lsp-request "textDocument/prepareRename"
+                               (lsp--text-document-position-params))))
         (-let* (((start . end) (lsp--range-to-region
-                                (if (lsp-range? response)
-                                    response
-                                  (lsp:prepare-rename-result-range response))))
-                (symbol (buffer-substring-no-properties start end))
-                (placeholder (lsp:prepare-rename-result-placeholder response)))
+                                (if (lsp-range? prepare)
+                                    prepare
+                                  (lsp:prepare-rename-result-range prepare))))
+                (symbol (buffer-substring start end))
+                (placeholder (lsp:prepare-rename-result-placeholder prepare)))
           (cons symbol (or placeholder symbol))))
-    (let ((symbol (thing-at-point 'symbol t)))
-      (cons symbol symbol))))
+    ;; We don't support prepare-rename, so just yield the symbol at point.
+    (let ((sym (thing-at-point 'symbol)))
+      (cons sym sym))))
```

Comment:
1. prepareRename does not work with `lsp-feature?`
2. Can you revert the renamed symbols?
3. Can you remove irrelevant comments?

Reply:
1. What should I use instead? The old code?
2, 3. Will do.

Reply:
The old code should be fine.

Reply:
Are you sure that it doesn't though? From stepping trough that with edebug, `lsp-feature?` returned a complex result for "textDocument/prepareRename", while consistently returning nil if I evaluate it somewhere without `lsp-mode` enabled (a *scratch* buffer, for instance).

Reply:
I checked and  lsp-method-requirements are actually implemented so lsp-feature? will work. It is odd that we haven't replaced the `lsp--get-symbol-to-rename`